rllib/utils/test_utils.py

```diff
@@ -435,40 +456,132 @@ def _test(
                     "for that!"
                 )
 
-    # Loop through: Policy vs Algorithm; Different API methods to calculate
-    # actions; unsquash option; clip option; full fetch or not.
-    for what in [pol, algorithm]:
+    def _test_compute_actions_from_raw_data(
+        what, method_to_test, explore, timestep, unsquash, clip
+    ):
         if what is algorithm:
-            # Get the obs-space from Workers.env (not Policy) due to possible
-            # pre-processor up front.
-            worker_set = getattr(algorithm, "workers", None)
-            assert worker_set
-            if isinstance(worker_set, list):
-                obs_space = algorithm.get_policy(pid).observation_space
-            else:
-                obs_space = worker_set.local_worker().for_policy(
-                    lambda p: p.observation_space, policy_id=pid
-                )
-            obs_space = getattr(obs_space, "original_space", obs_space)
+            return True
+
+        obs_space = getattr(
+            pol.observation_space, "original_space", pol.observation_space
+        )
+        obs = obs_space.sample()
+        if isinstance(obs_space, Box):
+            obs = np.clip(obs, -1.0, 1.0)
+
+        if method_to_test == "input_dict":
+            action, _, _ = pol.compute_actions_from_raw_input_dict(
```

Comment:
could u just use compute_actions_from_raw_inputs() directly here?

Reply:
raw_input_dict will get deprecated in the future, right?

Reply:
This PR only makes it so that this test_util works when connectors are switched off by default.
i.e. algorithm was trained before already and we only instantiate connectors spontaneously here to see if compute_actions_from_raw_input_dict works at all (in addition to making sure that compute_actions_from_input_dict works).

I have a follow up PR that also modifies algorithm to make this test_util also work if connectors were switched on while training and algorithm has no preprocessor.

Reply:
Tried to keep this PR a little smaller on the last mile. 🥈